### PR TITLE
refactor(power action): split power actions into 2 api's

### PR DESCRIPTION
**Pull request by brianosburn-intel**

### CODE REVIEW

File: src/routes/amt/index.ts [hunk 1]
```diff
@@ -25,6 +26,7 @@ amtRouter.get('/log/event/:guid', eventLog)
 amtRouter.get('/generalSettings/:guid', generalSettings)
 amtRouter.get('/hardwareInfo/:guid', hardwareInfo)
 amtRouter.post('/power/action/:guid', powerActionValidator(), powerAction)
+amtRouter.post('/power/action/extended/:guid', powerActionExtendedValidator(), powerActionExtended)
```
Comment (rsdmike) [inline src/routes/amt/index.ts:29]: what do you think of `/power/boot/:guid` ? or `/power/bootoptions/:guid`? @matt-primrose @graikhel-intel

**Review by rsdmike — CHANGES_REQUESTED** [review]
Please update commit to indicate breaking change, and associate the task number to this pr in the commit footer.

File: src/routes/amt/index.ts (hunk 1 above)
Comment (graikhel-intel) [reply src/routes/amt/index.ts:29]: I think /power/bootoptions/:guid makes more sense

File: src/routes/amt/powerAction.ts [hunk 2]
```diff
@@ -37,37 +37,35 @@ export async function powerAction (req: Request, res: Response): Promise<void> {
...
+    r.LockKeyboard = false
+    r.LockPowerButton = false
+    r.LockResetButton = false
+    r.LockSleepButton = false
+    r.ReflashBIOS = false
+    r.UseIDER = false
+    r.UseSOL = false
+    r.UseSafeMode = false
+    r.UserPasswordBypass = false
+    if (r.SecureErase) {
```
[the remark this answers is not in the document]
Comment (brianosburn-intel) [reply src/routes/amt/powerAction.ts:65]: @graikhel-intel suggests leaving it this way to workaround older firmware versions

**Review by rsdmike — APPROVED** [review]

**Pull request merged**